diff --git a/Makefile b/Makefile
index 088c344..71fc75e 100644
--- a/Makefile
+++ b/Makefile
@@ -54,6 +54,19 @@ filter_out2 = $(call filter_out1,$(call filter_out1,$1))
 unixpath = $(subst \,/,$1)
 unixcygpath = /$(subst :,,$(call unixpath,$1))
 
+# Minime
+ifeq ($(platform), minime)
+	override platform += unix
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	fpic := -fPIC
+	SHARED := -shared -Wl,-version-script=$(LIBRETRO_DIR)/link.T -Wl,-no-undefined
+	FLAGS += -fomit-frame-pointer -ffast-math -march=armv8-a+crc
+	FRONTEND_SUPPORTS_XRGB8888 = 1
+endif
+
 # Unix
 ifneq (,$(findstring unix,$(platform)))
 	fpic := -fPIC
