diff --git a/src/os/libretro/Makefile b/src/os/libretro/Makefile
index 0b5f0cd..2a1dcae 100644
--- a/src/os/libretro/Makefile
+++ b/src/os/libretro/Makefile
@@ -58,8 +58,19 @@ ifneq ($(GIT_VERSION)," unknown")
 endif
 
 GET_STATIC_ARG=$(if $(filter $(shell $(CXX) -print-file-name=$1.a),$1.a),,-static-$1)
+# Minime
+ifeq ($(platform), minime)
+   TARGET := stella_libretro.so
+   CC = $(CROSS_COMPILE)gcc
+   CXX = $(CROSS_COMPILE)g++
+   AR = $(CROSS_COMPILE)ar
+   fpic := -fPIC
+   SHARED := -shared -Wl,--version-script=link.T -Wl,--no-undefined
+   CXXFLAGS += -Ofast -fomit-frame-pointer -ffast-math -march=armv8-a+crc -std=c++17 -fno-rtti
+   LIBS := -lm
+
 # Unix
-ifneq (,$(findstring unix,$(platform)))
+else ifneq (,$(findstring unix,$(platform)))
    CXXFLAGS += $(LTO)
    LDFLAGS += $(LTO) $(PTHREAD_FLAGS) $(call GET_STATIC_ARG,libgcc) $(call GET_STATIC_ARG,libstdc++)
    TARGET := $(TARGET_NAME)_libretro.so
