diff --git a/libgambatte/libretro/libretro.cpp b/libgambatte/libretro/libretro.cpp
index 15b042c..2fa832e 100644
--- a/libgambatte/libretro/libretro.cpp
+++ b/libgambatte/libretro/libretro.cpp
@@ -2626,6 +2626,14 @@ static void check_variables(bool startup)
          {
             rgb32 = gb.gbcToRgb32(gbc_bios_palette[palnum * 4 + colornum]);
             gb.setDmgPaletteColor(palnum, colornum, rgb32);
+			
+			if (palnum==0 && colornum==0) {
+			    FILE *f = fopen("/tmp/dmg_grid_color", "w");
+			    if (f) {
+			        fprintf(f, "%u\n", rgb32);
+			        fclose(f);
+			    }
+			}
          }
       }
    }
