diff --git forkSrcPrefix/Makefile.libretro forkDstPrefix/Makefile.libretro
index 4560aacc..12abee77 100644
--- forkSrcPrefix/Makefile.libretro
+++ forkDstPrefix/Makefile.libretro
@@ -597,6 +597,15 @@ else ifeq ($(platform), miyoo)
 	DYNAREC = ari64

 # Emscripten
+# Minime
+else ifeq ($(platform), minime)
+   TARGET := $(TARGET_NAME)_libretro.so
+   CC = $(CROSS_COMPILE)gcc
+   CXX = $(CROSS_COMPILE)g++
+   AR = $(CROSS_COMPILE)ar
+   ARCH = arm64
+   DYNAREC = ari64
+
 else ifeq ($(platform), emscripten)
    TARGET  := $(TARGET_NAME)_libretro_$(platform).bc
    fpic    := -fPIC
