diff --git forkSrcPrefix/Makefile forkDstPrefix/Makefile
index a8b6835..483f082 100644
--- forkSrcPrefix/Makefile
+++ forkDstPrefix/Makefile
@@ -543,6 +543,18 @@ else ifneq (,$(findstring windows_msvc2017,$(platform)))
 	PSS_STYLE :=2
 	LDFLAGS += -DLL

+# Minime
+else ifeq ($(platform), minime)
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC  = $(CROSS_COMPILE)gcc
+   	CXX = $(CROSS_COMPILE)g++
+   	AR  = $(CROSS_COMPILE)ar
+	fpic := -fPIC
+	SHARED := -shared -Wl,--version-script=link.T
+	LIBM :=
+	FLAGS += -fomit-frame-pointer -ffast-math -march=armv8-a+crc
+	FLAGS += -DFAST_ALIGNED_LSB_WORD_ACCESS -fno-unroll-loops
+
 else
 	TARGET := $(TARGET_NAME)_libretro.dll
 	CC ?= gcc
