--- a/drivers/gpu/drm/panel/panel-mipi.c	2026-05-28 12:58:58.483500208 +0300
+++ b/drivers/gpu/drm/panel/panel-mipi.c	2026-05-28 12:59:10.697869874 +0300
@@ -1109,6 +1109,7 @@
 
 static const struct of_device_id panel_mipi_spi_of_match[] = {
 	{ .compatible = "panel-mipi-dpi-spi" },
+	{ .compatible = "anbernic,rg35xx-plus-panel" },
 	{ /* sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, panel_mipi_spi_of_match);
